### DISCUSSION TITLE
ST7735_GREENTAB128 rotation problem

### DISCUSSION THREAD

**Discussion by MHz000** [Bodmer/TFT_eSPI#2896] [New processor/display support or capability request]
As shown in the pictures the display works well. But there is a problem with rotation=1 and 0 (yellow and green background). Obviously there is an offset in the coordinate system. Can someone tell me how I can correct this?
Maybe a look at the platformio,ini or Setup code will help.
![rotation_2](https://github.com/Bodmer/TFT_eSPI/assets/7973146/aabec265-d3af-4b2b-925f-7e586372728f)
![rotation 3](https://github.com/Bodmer/TFT_eSPI/assets/7973146/663bfdfa-79fc-4897-a3a0-2f1b932cf395)
![rotation 1](https://github.com/Bodmer/TFT_eSPI/assets/7973146/b333397f-df45-4d1c-8311-e18ca9e20324)
![rotation 0](https://github.com/Bodmer/TFT_eSPI/assets/7973146/b079b010-d1ed-4f01-b55c-1d996bbcca2f)
platformio.ini
<code>
[env:d1]
platform = espressif8266
board = d1
framework = arduino
monitor_speed = 115200
upload_speed = 512000
build_flags = 
	-D USER_SETUP_LOADED=1
	-D ST7735_DRIVER=1
	-D TFT_WIDTH=128
	-D TFT_HEIGHT=128
	-D TFT_CS=PIN_D8
	-D TFT_DC=PIN_D3
	-D TFT_RST=PIN_D4
	-D LOAD_GLCD=1
	-D LOAD_FONT2=1
	-D LOAD_FONT4=1
	-D LOAD_FONT6=1
	-D LOAD_FONT7=1
	-D LOAD_FONT8=1
	-D SPI_FREQUENCY=27000000
lib_deps = bodmer/TFT_eSPI@^2.5.31
</code>

Setup7_ST7735_128x128
<code>
#define USER_SETUP_ID 7
#define ST7735_DRIVER
#define TFT_WIDTH  128
#define TFT_HEIGHT 128
#define ST7735_GREENTAB128 // For 128 x 128 display  
#define TFT_CS   PIN_D8  // Chip select control pin D8
#define TFT_DC   PIN_D3  // Data Command control pin
#define TFT_RST  PIN_D4  // Reset pin (could connect to NodeMCU RST, see next line)
#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT
#define SPI_FREQUENCY  27000000
#define SPI_TOUCH_FREQUENCY  2500000
</code>

**Comment by Bodmer** [comment]
The offset correction line has been missed out, add:


-D ST7735_GREENTAB128=1

**Comment by MHz000** [comment]
@Bodmer
Exactly, that was the missing definition. Thank you for this hint. The display now works perfectly. :-))

Is there a document where I could have found out myself that this line was missing in my User_Setup. In which folder TFT_Drivers, xxx_Defines, xxx_Init, xxxRotation etc can I find the missing line #define ST7735_GREENTAB128=1?

I have to admit that I don't understand the procedure for unknown displays.
Thank you for your patience

**Reply by Azhilus** [reply]
Hey @MHz000 can you guide me on how you got it running because I am having difficulties running it 
![WhatsApp Image 2024-01-22 at 10 52 15](https://github.com/Bodmer/TFT_eSPI/assets/66466976/313039ec-fe52-4eb3-9258-042e60b09370)
```ino
#include <TFT_eSPI.h>
#include <SPI.h>
TFT_eSPI tft = TFT_eSPI();

void setup() {
  // put your setup code here, to run once:
  tft.init();
  tft.fillScreen(0x0c70);
}

void loop() {
  // put your main code here, to run repeatedly:

}
```
